utils/testrunner/test/testrunner.spec.js

```diff
@@ -0,0 +1,522 @@
+        await new Promise(x => setTimeout(x, 10));
+        ran = true;
+      });
+      await t.run();
+      expect(ran).toBe(true);
+    });
+    it('should run async tests in order of their declaration', async() => {
+      const log = [];
+      const t = new TestRunner();
+      t.it('uno', async() => {
+        await new Promise(x => setTimeout(x, 30));
+        log.push(1);
+      });
+      t.it('dos', async() => {
+        await new Promise(x => setTimeout(x, 20));
+        log.push(2);
+      });
+      t.it('tres', async() => {
+        await new Promise(x => setTimeout(x, 10));
+        log.push(3);
+      });
+      await t.run();
+      expect(log.join()).toBe('1,2,3');
+    });
+    it('should run multiple tests', async() => {
+      const log = [];
+      const t = new TestRunner();
+      t.it('uno', () => log.push(1));
+      t.it('dos', () => log.push(2));
+      await t.run();
+      expect(log.join()).toBe('1,2');
+    });
+    it('should NOT run a skipped test', async() => {
+      const t = new TestRunner();
+      let ran = false;
+      t.xit('uno', () => ran = true);
+      await t.run();
+      expect(ran).toBe(false);
+    });
+    it('should run ONLY non-skipped tests', async() => {
+      const log = [];
+      const t = new TestRunner();
+      t.it('uno', () => log.push(1));
+      t.xit('dos', () => log.push(2));
+      t.it('tres', () => log.push(3));
+      await t.run();
+      expect(log.join()).toBe('1,3');
+    });
+    it('should run ONLY focused tests', async() => {
+      const log = [];
+      const t = new TestRunner();
+      t.it('uno', () => log.push(1));
+      t.xit('dos', () => log.push(2));
+      t.fit('tres', () => log.push(3));
+      await t.run();
+      expect(log.join()).toBe('3');
+    });
+    it('should run tests in order of their declaration', async() => {
+      const log = [];
+      const t = new TestRunner();
+      t.it('uno', () => log.push(1));
+      t.describe('suite1', () => {
+        t.it('dos', () => log.push(2));
+        t.it('tres', () => log.push(3));
+      });
+      t.it('cuatro', () => log.push(4));
+      await t.run();
+      expect(log.join()).toBe('1,2,3,4');
+    });
+    it('should support async test suites', async() => {
+      const log = [];
+      const t = new TestRunner();
+      t.it('uno', () => log.push(1));
+      await t.describe('suite1', async() => {
+        await Promise.resolve();
+        t.it('dos', () => log.push(2));
+        await Promise.resolve();
+        t.it('tres', () => log.push(3));
+      });
+      t.it('cuatro', () => log.push(4));
+      await t.run();
+      expect(log.join()).toBe('1,2,3,4');
+    });
+  });
+
+  describe('TestRunner parallel', () => {
+    it('should run tests in parallel', async() => {
+      const log = [];
+      const t = new TestRunner({parallel: 2});
+      t.it('uno', async state => {
+        log.push(`Worker #${state.parallelIndex} Starting: UNO`);
+        await Promise.resolve();
```

Comment:
up to you, but relying on the number of ticks that await Promise.resolve() takes seems like a good way to get burned.

Reply:
Yeah I was hesitant on this one as well. Let's wait to get burned and iterate from there